aws/ec2.go

```diff
@@ -110,3 +113,394 @@ func nukeAllEc2Instances(session *session.Session, instanceIds []*string) error
 	logging.Logger.Infof("[OK] %d instance(s) terminated in %s", len(instanceIds), *session.Config.Region)
 	return nil
 }
+
+func GetEc2ServiceClient(region string) ec2iface.EC2API {
+	return ec2.New(newSession(region))
+}
+
+type Vpc struct {
+	Region string
+	VpcId  string
+	svc    ec2iface.EC2API
+}
+
+func NewVpcPerRegion(regions []string) []Vpc {
+	var vpcs []Vpc
+	for _, region := range regions {
+		vpc := Vpc{
+			svc:    GetEc2ServiceClient(region),
+			Region: region,
+		}
+		vpcs = append(vpcs, vpc)
+	}
+	return vpcs
+}
+
+func GetDefaultVpcId(vpc Vpc) (string, error) {
+	input := &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   awsgo.String("isDefault"),
+				Values: []*string{awsgo.String("true")},
+			},
+		},
+	}
+	vpcs, err := vpc.svc.DescribeVpcs(input)
+	if err != nil {
+		logging.Logger.Errorf("[Failed] %s", err)
+		return "", errors.WithStackTrace(err)
+	}
+	if len(vpcs.Vpcs) == 1 {
+		return awsgo.StringValue(vpcs.Vpcs[0].VpcId), nil
+	} else if len(vpcs.Vpcs) > 1 {
+		// More than one VPC in a region should never happen
+		err = fmt.Errorf("Impossible - more than one VPC found in region %s", vpc.Region)
+		return "", errors.WithStackTrace(err)
+	}
+	// No default VPC
+	return "", nil
+}
+
+func GetDefaultVpcs(vpcs []Vpc) ([]Vpc, error) {
+	for i := 0; i < len(vpcs); i++ {
+		vpcId, err := GetDefaultVpcId(vpcs[i])
+		if err != nil {
+			return []Vpc{Vpc{}}, errors.WithStackTrace(err)
+		}
+		if vpcId != "" {
+			vpcs[i].VpcId = vpcId
+		} else {
+			// Ignore regions that don't have a default VPC
+			vpcs = append(vpcs[:i], vpcs[i+1:]...)
+			i--
+		}
+	}
+	return vpcs, nil
```

Comment:
Also, I am confused about the input var. Is that all VPCs that this has detected, or is the input supposed to be one example VPC per region?

Reply:
The input var is really just a `Vpc{}` object for each region with the `Region` and `svc` fields already assigned. The reason to do it this way (and it does feel a bit funky to me) is that the `svc` object is overridden by the mock object in the tests. in the CLI we call `NewVpcPerRegion()` followed by `GetDefaultVpcs()`. The tests skip the `NewVpcPerRegion()` (which assigns `svc` to the "real" service object and instead assigns the mock object itself.

Reply:
Re: the loop above - nice catch. This was an experiment I had neglected to replace. I'll go with something similar to your suggestion.

Reply:
> The reason to do it this way (and it does feel a bit funky to me) is that the svc object is overridden by the mock object in the tests.

Ah ok. Probably worth adding an extended comment there since I am bound to come in and try to "optimize" this at a later point in time and shoot myself in the foot!